src/test/java/uk/gov/hmcts/juror/api/moj/report/grouped/CompletionOfServiceReportTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.juror.api.TestConstants.COURT_NAME_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.COURT_NAME_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_FROM_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_FROM_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_TO_DISPLAY_NAME;
+import static uk.gov.hmcts.juror.api.TestConstants.DATE_TO_KEY;
+import static uk.gov.hmcts.juror.api.TestConstants.VALID_COURT_LOCATION;
+
+class CompletionOfServiceReportTest extends AbstractGroupedReportTestSupport<CompletionOfServiceReport> {
+
+    private static final LocalDate DATE_FROM_TEST_VALUE = LocalDate.of(2023, 3, 1);
+    private static final LocalDate DATE_TO_TEST_VALUE = LocalDate.of(2023, 3, 2);
+
+    private MockedStatic<SecurityUtil> securityUtilMockedStatic;
+    private CourtLocationService courtLocationService;
+
+    public CompletionOfServiceReportTest() {
+        super(
+            QJurorPool.jurorPool,
+            CompletionOfServiceReport.RequestValidator.class,
+            ReportGroupBy.builder()
+                .dataType(DataType.POOL_NUMBER_AND_COURT_TYPE)
+                .removeGroupByFromResponse(true)
+                .build(),
+            DataType.JUROR_NUMBER,
+            DataType.FIRST_NAME,
+            DataType.LAST_NAME,
+            DataType.COMPLETION_DATE
+        );
+    }
+
+    @BeforeEach
+    @Override
+    public void beforeEach() {
+        this.securityUtilMockedStatic = mockStatic(SecurityUtil.class);
+        this.courtLocationService = mock(CourtLocationService.class);
+        super.beforeEach();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        securityUtilMockedStatic.close();
+    }
+
+
+    @Override
+    public CompletionOfServiceReport createReport(PoolRequestRepository poolRequestRepository) {
+        return new CompletionOfServiceReport(poolRequestRepository, courtLocationService);
+    }
+
+    @Override
+    protected StandardReportRequest getValidRequest() {
+        return StandardReportRequest.builder()
+            .reportType(report.getName())
+            .fromDate(DATE_FROM_TEST_VALUE)
+            .toDate(DATE_TO_TEST_VALUE)
+            .build();
+    }
+
+    @Override
+    public void positivePreProcessQueryTypical(JPAQuery<Tuple> query, StandardReportRequest request) {
+        securityUtilMockedStatic.when(SecurityUtil::isCourt).thenReturn(true);
+        securityUtilMockedStatic.when(SecurityUtil::getLocCode).thenReturn(VALID_COURT_LOCATION);
+        report.preProcessQuery(query,request);
+
+        verify(query).where(QJurorPool.jurorPool.juror.completionDate.between(DATE_FROM_TEST_VALUE,
+            DATE_TO_TEST_VALUE));
+        verify(query).where(QJurorPool.jurorPool.location.eq(SecurityUtil.getLocCode()));
```

Comment:
SecurityUtil.getLocCode() should be the expected value instead of calling the method

Reply:
Resolved!